src/main/java/com/google/finapp/SpannerDaoImpl.java

```diff
@@ -55,6 +60,9 @@ public void createCustomer(ByteArray customerId, String name, String address)
   public void createAccount(
       ByteArray accountId, AccountType accountType, AccountStatus accountStatus, BigDecimal balance)
       throws SpannerDaoException {
+    if (balance.signum() == -1) {
+      throw new IllegalArgumentException("Account balance cannot be negative");
```

Comment:
Optional: You could also put the negative balance that caused the issue in the exception string, this could help with debugging in a real application. Similar comment for all other exception strings.